src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -363,11 +363,11 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
   // Note:
   // we take the most significant bits of the hashcode, not the lower ones
   // this is of crucial importance when populating the table in parallel
+  // See: https://graphics.stanford.edu/~seander/bithacks.html#IntegerLogDeBruijn for explanation and BitManipulationBenchmark for alternatives
+  private val exponents = Array(32, 31, 4, 30, 3, 18, 8, 29, 2, 10, 12, 17, 7, 15, 28, 24, 1, 5, 19, 9, 11, 13, 16, 25, 6, 20, 14, 26, 21, 27, 22, 23)
   protected final def index(hcode: Int) = {
-    val ones = table.length - 1
-    val improved = improve(hcode, seedvalue)
-    val shifted = (improved >> (32 - java.lang.Integer.bitCount(ones))) & ones
-    shifted
+    val exponent = exponents((table.length * 0x077CB531) >>> 27)
+    (improve(hcode, seedvalue) >>> exponent) & (table.length - 1)
```

Comment:
> Yes, please check out the benchmarks below:

Oops, I had actually had a look at the benchmarks, but somehow my eyes did not see the one I was looking for ^^